### DISCUSSION TITLE
Missing start button ?

### DISCUSSION THREAD

**Discussion by neurodavidus** [wehr-lab/autopilot#81] [Q&A]
Hi there
After creating a protocol and a subject, I don't see how to start the task.
Maybe something is wrong with the GUI ? this is how it looks :

![image](https://user-images.githubusercontent.com/4154936/108879534-a928cb80-7601-11eb-96e8-e03dbc54b3c0.png)

Any suggestion?
thanks a lot. DaVID

**Comment by sneakers-the-rat** [comment]
The start button will be grayed out until the Terminal receives a message from the Pilot indicating it's alive and has made contact. The terminal will be looking for a message from a pilot with an id that matches the one in the pilot registry -- i looks like you've given the ip address of the pilot in the `pilot_db.json` file (which is used to populate the gui), that should be whatever you have in the `NAME` field of `prefs.json` on the pilot. So I think you would need to edit the `pilot_db.json` file to reflect that. If you're still having trouble connecting the two agents, please set the `LOGLEVEL` of both to `DEBUG` , zip and upload the logs (in `~/autopilot/logs`) as well as the `prefs.json` files and i'll take a look.

Come to think of it the GUI should add another slot anytime a pilot makes contact automatically rather than needing to be added manually...

**Reply by neurodavidus** [reply]
Thanks Jonny. 
So I did the change you suggested. Now the GUI shows the name of the pilot instead of its IP. But still no green start button.
I went back to the installation instruction and saw this line : 
So, if connecting a Pilot to a Terminal, the PUSHPORT of the Pilot should match the MSGPORT of the Terminal.
I noticed that by default there is a mismatch. the terminal MSGPORT is  5565 and the pilot PUSHPORT is 5560.  I fixed this but still no luck. 
Then by some kind of miracle, the green button appeared. So it seems that the order by which you start autopilot on the terminal and pilot matters! To get the green button you need to launch autopilot on the terminal **then** launch it on the pilot.

Unfortunately, I did not manage to blink the LED using the blink task (nothing happen on the PI when I start the blink task). 

When I press the start button, the terminal asks me the weight of my animal. Some data are saved when I stop the task. But the LED does not blink. I tested my electronic set-up using a simple python code on the raspberry so I know there is no error there. I guess the problem is setting the hardware correctly. I tried to follow the instruction you gave me (I added the `HARDWARE` dictionary within `prefs.json` in the pilot). Maybe I missed a step in the configuration of the hardware on the pilot.  Anyway getting slowly there but I think there is a need for a _step-by-step_ tutorial for dummies: how to install autopilot and run a blinking task". When I have managed to make it work I can take care of that ...

thanks again for the fast help ...David

**Reply by sneakers-the-rat** [reply]
Sorry for the delay, been busy and then ill. 

Would it be possible for you to set `LOGLEVEL` to `DEBUG` in the `prefs.json` file for the pilot and the terminal, run the task, and post the logs? That will let me debug it, and should also give you a much clearer picture of what's going on :) (it's at `WARNING` by default for performance because printing to stdout and writing to file actually takes some time)

Only thing that comes to mind is that perhaps you entered the BCM numbering of the pin you're using, as opposed to the board position of the pin? So for example looking at this pinout: https://www.raspberrypi-spy.co.uk/wp-content/uploads/2012/06/Raspberry-Pi-GPIO-Header-with-Photo-1078x516.png GPIO number 4 in the BCM scheme (the `GPIO4` text) is at board position 7 (circled number 7) -- autopilot uses the board number (7)

**Reply by sneakers-the-rat** [reply]
and you're correct re: the port numbering, it's an unfortunate quirk of the defaults system and one that i hope will be obsolete when i rewrite the network discovery system.

unfortunately many things to improve, especially the docs! but i'm being pulled between my PhD program requirements, a few collaborations, and tending to bugfixes so will hopefully have more time to address your very helpful comments as soon as the degree requirements are off my plate :)

**Reply by neurodavidus** [reply]
Hi Jonny, my turn to have been busy.
So in fact a while ago (and in fact before you answered me) I manage to get the blinking task to work !! (yeah!!)
There were several reasons why it did not work. First during the installation procedure, because I did not know what I was going to do, the pilot hardware was not well configurated. I rerun the configuration script, added the hardware (select one pin from GPIO and name it according to your advice). Then I modified my blink task (blink.py) by declaring the hardware according to the name I chose in the previous step and of course not forgetting to change the pulse function accordingly. At this step, when I started the task on the terminal, I saw an error message on the PI which made me realize that I needed to have, on the PI, the blink.py task properly set up like on the terminal (somehow I had not understood this and was thinking that tasks only really run on the terminal, not on the PI). At this stage, it was still not working but I had no more error. I played with the protocol and realize that I needed to fill in whatever information in the graduation criterion. And the magic occurred. The LED blinked.  So now I am going to simulate something a bit more complex (maybe play with your go-nogo task). And next step will be the camera ....
If there are things that make no sense in my above comments let me know.  maybe I will post a few additional questions on things that are not entirely clear for me.
Thanks
David

**Reply by sneakers-the-rat** [reply]
Ah! yes! glad you got it working :):) the graduation criteria shouldn't be mandatory, thanks for flagging that for me!
